File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
  let answer = 1;
    let bridge = new Array(bridge_length).fill(0);
    let bridgr_sum = 0;
    
    
    bridge.shift();
    bridgr_sum += truck_weights[0];
    bridge.push(truck_weights.shift());
    
    while(bridgr_sum > 0){
        answer++;
        
        bridgr_sum -= bridge.shift();
        if(truck_weights.length > 0 && bridgr_sum + truck_weights[0] <= weight){
            bridgr_sum += truck_weights[0];
            bridge.push(truck_weights.shift());
        }
        else{
            bridge.push(0);
        }
    }
  
  return answer;
}

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    time = 0
    
    bridge = [0] * bridge_length
    current_weight = 0
    
    while current_weight > 0 or truck_weights:
        time += 1
        current_weight -= bridge.pop(0)
        
        if truck_weights:
            next_truck = truck_weights[0]
            
            if current_weight + next_truck <= weight:
                current_weight += truck_weights[0]
                bridge.append(truck_weights[0])
                truck_weights.pop(0)
            else:
                bridge.append(0)
        else:
            bridge.append(0)
            
    return time
            
    